cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/ProgramStatusTrigger.java

```diff
@@ -39,4 +47,40 @@ public ProgramStatusTrigger(ProgramId programId, Set<ProgramStatus> programStatu
   public ProgramStatusTrigger(ProgramId programId, ProgramStatus... programStatuses) {
     super(programId, new HashSet<>(Arrays.asList(programStatuses)));
   }
+
+  @Override
+  public boolean updateStatus(Notification notification) {
+    satisfied = true;
+    return satisfied;
+  }
+
+  @Override
+  public boolean isSatisfied() {
+    return satisfied;
+  }
+
+  @Override
+  public List<String> getTriggerKeys() {
+    return ImmutableList.of(programId.toString());
```

Comment:
The trigger key is based on the program status too - so it's more than just `programId`.

For every `programStatus`, it's:
```
programId.toString() + "." + programStatus.toString().toLowerCase();
```